src/lib/openapi/spec/addon-create-update-schema.ts

```diff
@@ -0,0 +1,73 @@
+import { FromSchema } from 'json-schema-to-ts';
+
+export const addonCreateUpdateSchema = {
+    $id: '#/components/schemas/addonCreateUpdateSchema',
+    type: 'object',
+    required: ['provider', 'enabled', 'parameters', 'events'],
+    properties: {
+        provider: {
+            type: 'string',
+
+            description: `The addon provider, such as "webhook" or "slack". The list of all providers supported can be found in the [addons reference documentation](https://docs.getunleash.io/reference/addons). This string is **case sensitive**. Supported options are:
+- \`datadog\` for [Datadog](https://docs.getunleash.io/reference/addons/datadog)
+- \`slack\` for [Slack](https://docs.getunleash.io/reference/addons/slack)
+- \`teams\` for [Microsoft Teams](https://docs.getunleash.io/reference/addons/teams)
+- \`webhook\` for [webhooks](https://docs.getunleash.io/reference/addons/webhook)
+
+The provider you choose for your addon dictates what properties the \`parameters\` object needs. Refer to the documentation for each provider for more information.
```

Comment:
We have very scant documentation on this, so until those are improved, I think putting this in here would be useful.

Reply:
Registered providers might refer to the endpoint in the /addons endpoint, where we can see all known providers for the instance. I guess this has exposed the need for fleshing out the addon documentation.